sycl/source/detail/graph_impl.cpp

```diff
@@ -378,83 +380,96 @@ sycl::event exec_graph_impl::enqueue(
       }
-      ScheduledEvents.push_back(NewEvent);
     } else {
-      auto EventImpl = sycl::detail::createCommandAndEnqueue(
-          NodeImpl->MCGType, Queue, NodeImpl->MNDRDesc,
-          nullptr /* HostKernel */, nullptr /* HostTaskPtr */,
-          nullptr /* InteropTask */, NodeImpl->MKernel, NodeImpl->MKernelName,
-          nullptr /* KernelBundle */, NodeImpl->MArgStorage,
-          NodeImpl->MAccStorage, NodeImpl->MLocalAccStorage,
-          NodeImpl->MStreamStorage, {} /* shared_ptr storage */,
-          NodeImpl->MAuxiliaryResources, NodeImpl->MArgs, nullptr /* SrcPtr */,
-          nullptr /* DstPtr */, 0 /* Length */, {} /* Pattern */,
-          0 /* SrcPitch */, 0 /* DstPitch */, 0 /* Width */, 0 /* Height */,
-          0 /* Offset */, false /* IsDeviceImageScoped */,
-          {} /* HostPipeName */, nullptr /* HostPipePtr */,
-          false /* HostPipeBlocking */, 0 /* HostPipeTypeSize */,
-          false /* HostPipeRead */, {} /* Advice */, NodeImpl->MRequirements,
-          {} /* Events */, {} /* Events w/ Barrier */,
-          NodeImpl->MOSModuleHandle,
-          PI_EXT_KERNEL_EXEC_INFO_CACHE_DEFAULT
-          /* KernelCacheConfig */,
-          {} /* CodeLoc */);
-
-      ScheduledEvents.push_back(EventImpl);
+      std::unique_ptr<sycl::detail::CG> CommandGroup =
+          std::make_unique<sycl::detail::CGExecCommandBuffer>(CommandBuffer,
+                                                              MRequirements);
+
+      NewEvent = sycl::detail::Scheduler::getInstance().addCG(
+          std::move(CommandGroup), Queue);
     }
+  } else {
+    std::vector<std::shared_ptr<sycl::detail::event_impl>> ScheduledEvents;
+    for (auto &NodeImpl : MSchedule) {
+      std::vector<RT::PiEvent> RawEvents;
+
+      // If the node has no requirements for accessors etc. then we skip the
+      // scheduler and enqueue directly.
+      if (NodeImpl->MCGType == sycl::detail::CG::Kernel &&
+          NodeImpl->MRequirements.size() + NodeImpl->MStreamStorage.size() ==
+              0) {
+        auto NewEvent = CreateNewEvent();
+        RT::PiEvent *OutEvent = &NewEvent->getHandleRef();
+        pi_int32 Res =
+            sycl::
+                detail::enqueueImpKernel(Queue, NodeImpl->MNDRDesc,
+                                         NodeImpl->MArgs,
+                                         nullptr /* TODO: Handle KernelBundles
+                                                  */
+                                         ,
+                                         NodeImpl->MKernel,
+                                         NodeImpl->MKernelName,
+                                         NodeImpl->MOSModuleHandle, RawEvents,
+                                         OutEvent,
+                                         nullptr /* TODO: Pass mem allocation
+                                                    func for accessors */
+                                         ,
```

Comment:
The formatting with these parameter comments seems to have gotten a bit messed up.

Reply:
Manually sorted this out, by moving comments to the line above